src/System.Management.Automation/engine/serialization.cs

```diff
@@ -3992,14 +3993,27 @@ private object ReadListContainer(ContainerType ct)
         /// Deserialize Dictionary.
         /// </summary>
         /// <returns></returns>
-        private object ReadDictionary(ContainerType ct)
+        private object ReadDictionary(ContainerType ct, ConsolidatedString InternalTypeNames)
         {
             Dbg.Assert(ct == ContainerType.Dictionary, "Unrecognized ContainerType enum");
 
             // We assume the hash table is a PowerShell hash table and hence uses
             // a case insensitive string comparer.  If we discover a key collision,
             // we'll revert back to the default comparer.
             Hashtable table = new Hashtable(StringComparer.CurrentCultureIgnoreCase);
+            OrderedDictionary tableOrdered = new OrderedDictionary(StringComparer.CurrentCultureIgnoreCase);
+
+            // Find whether original directory was odered or not
+            bool isOrdered = false;
+            if (InternalTypeNames.Count > 0)
+            {
+                string originalDictTypeName = Deserializer.MaskDeserializationPrefix(InternalTypeNames[0]);
+                if (originalDictTypeName == tableOrdered.GetType().FullName)
+                {
+                    isOrdered = true;
+                }
+            }
```

Comment:
There is a typo in comment - "odered".

Also we could pack the code in one line initialization:
```suggestion
            // Find whether original directory was ordered or not
            bool isOrdered = InternalTypeNames.Count > 0 && (Deserializer.MaskDeserializationPrefix(InternalTypeNames[0]) == tableOrdered.GetType().FullName));
```

Reply:
Changed.